client/client.c

```diff
@@ -0,0 +1,146 @@
+#include "client.h"
+
+#define CHUNK_SIZE 512
+
+//Receive data in multiple chunks by checking a non-blocking socket
+//Timeout in seconds
+int recv_timeout(int s, int timeout)
+{
+    int size_recv, total_size = 0;
+    struct timeval begin;
+    struct timeval now;
+    char chunk[CHUNK_SIZE];
+    double timediff;
+
+    //make socket non blocking
+    fcntl(s, F_SETFL, O_NONBLOCK);
+
+    //beginning time
+    gettimeofday(&begin, NULL);
+
+    while (1)
+    {
+        gettimeofday(&now, NULL);
+
+        //time elapsed in seconds
+        timediff = (now.tv_sec - begin.tv_sec) + 1e-6 * (now.tv_usec - begin.tv_usec);
+
+        //if you got some data, then break after timeout
+        if (total_size > 0 && timediff > timeout)
+        {
+            break;
+        }
+
+        //if you got no data at all, wait a little longer, twice the timeout
+        else if (timediff > timeout * 2)
+        {
+            break;
+        }
+
+        memset(chunk, 0, CHUNK_SIZE); //clear the variable
+        if ((size_recv = recv(s, chunk, CHUNK_SIZE, 0)) < 0)
+        {
+            //if nothing was received then we want to wait a little before trying again, 0.1 seconds
+            usleep(100000);
+        }
+        else
+        {
+            total_size += size_recv;
+            printf("%s", chunk);
+            //reset beginning time
+            gettimeofday(&begin, NULL);
+        }
+    }
+    return total_size;
+}
+
+void *clientThread(void *arg)
+{
+    short port = 8080;
```

Comment:
Use a less generic port number to not have possible conflicts with the system.

Reply:
Used 49152 port address.